low-level/files/opt/seagate/sspl/setup/resource_map/storage.py

```diff
@@ -204,6 +209,91 @@ def build_encl_platform_sensors_data(self, platform_sensors):
                             sensors_data[platform_sensor] = [single_sensor_data]
         return sensors_data
 
+    def get_logical_volumes_info(self):
+        """Update and return logical volume information in specific format."""
+        logvol_data = []
+        logicalvolumes = self.get_realstor_encl_data("volumes")
+        if logicalvolumes:
+            for logicalvolume in logicalvolumes:
+                health = logicalvolume.get("health", "NA")
+                if health in HEALTH_UNDESIRED_VALS:
+                    health = "NA"
+                logvol_data.append({
+                    "uid": logicalvolume.get("volume-name", "NA"),
+                    "fru": False,
+                    "last_updated": int(time.time()),
+                    "health": {
+                        "status": health,
+                        "description": logicalvolume.get("volume-description", "NA"),
+                        "recommendation": logicalvolume.get("health-recommendation", "NA"),
+                        "specifics": [
+                            {
+                                "disk_group": {
+                                    "disk_group_name": logicalvolume.get("container-name", "NA"),
+                                    "pool_serial_number": logicalvolume.get("container-serial", "NA")
+                                }
+                            }
+                        ]
+                    }
+                })
+        return logvol_data
+
+    def get_disk_groups_info(self):
+        """Update and return disk-group information in specific format."""
+        dg_data = []
+        dg_vol_map = {}
+        diskgroups = self.get_realstor_encl_data("disk-groups")
+        # Mapping logical volumes with disk group.
+        logicalvolumes = self.get_realstor_encl_data("volumes")
+        if logicalvolumes:
+            for logicalvolume in logicalvolumes:
+                volume_pool_sr_no = logicalvolume.get("container-serial", "NA")
+                volume_uid = logicalvolume.get("volume-name", "NA")
+                if volume_pool_sr_no in dg_vol_map:
+                    dg_vol_map[volume_pool_sr_no].append({"volume_uid": volume_uid})
+                else:
+                    dg_vol_map.update({volume_pool_sr_no:[{"volume_uid": volume_uid}]})
+        dg_description = "Disk group is in good health."
+        if diskgroups:
+            for diskgroup in diskgroups:
+                health = diskgroup.get("health", "NA")
+                pool_sr_no = diskgroup.get("pool-serial-number", "NA")
+                if pool_sr_no in dg_vol_map:
+                    volumes = dg_vol_map[pool_sr_no]
+                else:
+                    volumes = None
+                if health in HEALTH_UNDESIRED_VALS:
```

Comment:
Where can I find dg_data which are in good health? Else part is missing.

Reply:
Actually, we will get this health info from the enclosure API itself.
In If case we are just validating if health is "N/A" or "" then that time we assume health is "NA" (this is a requirement that we got previously from CSM that health field is only included with "NA", "Fault" or "Degraded")

Reply:
Thats fine, @satish-seagate. What I'm interested on is where we are collecting the healthy dg_data?
Here I can see dg_data.append() takes only unhealthy instances.

Reply:
Ok, got it its indented outside if.